### DISCUSSION TITLE
Upptime-bot is deleting gh-pages

### DISCUSSION THREAD

**Discussion by redlinejoes** [upptime/upptime#718] [Q&A]
Repo is using GitHub Pages
Repo is public

upptime bot is deleting the content of the gh-pages branch when it deploys a commit. 

Why does this happen?

How can this be resolved while maintaining the upptime bot's functionality? 

The other discussion on this topic was set to "resolved" prematurely, and none of the suggestions apply to the specific issue.

**Comment by AnandChowdhary** [comment]
Can you please share a repository and I'll have a look? :)

**Reply by redlinejoes** [reply]
The issue must be related to incomplete configuration YAML. When configuring upptime for the first time, if you do not complete the YAML configuration file customization, the bot will wipe your gh-pages branch. 

You can confirm this by creating a new repo using the template and then don't make any change to the configuration YAML. After the upptime bot starts pushing commits, it will delete the gh-pages content.

**Reply by redlinejoes** [reply]
Failure to complete the YAML configuration by the user should be marked as the cause of the issue. 

The workaround is to complete the necessary YAML configuration for upptime before the CI bot has a chance to wipe the gh-pages branch content.

**Comment by redlinejoes** [comment]
Github.com/redlinejoes/upptime

I wiped the repo and started from scratch. Let's give it about 6hrs and
see what happens. Thanks for reaching out so fast.

I have upptime fully configured with a public repo, and custom domain,  and
CICD appears to be working now. I suspect the issue is related to
incomplete configuration of the YAML file.

When it was deleting data, I had not completed making configuration changes
to the YAML config file.
